dnf5-plugins/config-manager_plugin/setvars.cpp

```diff
@@ -0,0 +1,105 @@
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "setvars.hpp"
+
+#include "shared.hpp"
+
+#include <libdnf5/utils/bgettext/bgettext-mark-domain.h>
+
+#include <fstream>
+
+namespace dnf5 {
+
+using namespace libdnf5::cli;
+
+void ConfigManagerCommandSetVars::set_argument_parser() {
+    auto & ctx = get_context();
+    auto & parser = ctx.get_argument_parser();
+
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_description("Set variables");
+
+    auto vars_vals =
+        parser.add_new_positional_arg("varvals", ArgumentParser::PositionalArg::AT_LEAST_ONE, nullptr, nullptr);
+    vars_vals->set_description("List of variables with values. Format: \"variable=value\"");
+    vars_vals->set_parse_hook_func(
+        [this, &ctx]([[maybe_unused]] ArgumentParser::PositionalArg * arg, int argc, const char * const argv[]) {
+            for (int i = 0; i < argc; ++i) {
+                auto value = argv[i];
+                auto val = strchr(value + 1, '=');
+                if (!val) {
+                    throw libdnf5::cli::ArgumentParserError(
+                        M_("varval: Badly formatted argument value \"{}\""), std::string{value});
+                }
+                std::string var_name{value, val};
+                std::string var_value{val + 1};
+
+                check_variable_name(var_name);
+
+                // Test that the variable is not read-only.
+                auto vars = ctx.base.get_vars();
+                if (vars->is_read_only(var_name)) {
+                    throw ConfigManagerError(
+                        M_("Cannot set \"{}\": Variable \"{}\" is read-only"), std::string{value}, var_name);
+                }
+
+                // Save the variable for later writing to a file.
+                const auto [it, inserted] = setvars.insert({var_name, var_value});
+                if (!inserted) {
+                    if (it->second != var_value) {
+                        throw ConfigManagerError(
+                            M_("Sets the \"{}\" variable again with a different value: \"{}\" != \"{}\""),
+                            var_name,
+                            it->second,
+                            var_value);
+                    }
+                }
+            }
+            return true;
+        });
+    cmd.register_positional_arg(vars_vals);
+}
+
+
+void ConfigManagerCommandSetVars::configure() {
+    auto & ctx = get_context();
+
```

Comment:
What about to add to all configure steps -> `ctx.set_load_system_repo(false);`. I think that loading RPMDB is not required by the plugin. What do you think?

Reply:
@j-mracek `load_system_repo` is `false` by default. So adding `ctx.set_load_system_repo(false);` does not change anything.
The plugin does not need RPMDB. If the RPMDB is created, it will be caused by something other than loading. Maybe librpm will do this during architecture detection? Maybe ... I don't know, I need to investigate.